src/evm/protocol/uniswap_v4/hooks/generic_vm_hook_handler.rs

```diff
@@ -223,23 +228,85 @@ where
 
     fn get_amount_ranges(
         &self,
-        _token_in: Bytes,
-        _token_out: Bytes,
+        token_in: Bytes,
+        token_out: Bytes,
     ) -> Result<AmountRanges, SimulationError> {
-        Err(SimulationError::RecoverableError(
-            "get_amount_ranges is not implemented for GenericVMHookHandler".to_string(),
-        ))
+        match &self.limits_entrypoint {
```

Comment:
`limits_entrypoint` is just an Option.. why do you prefer to use `match` over `if let Some(..)=`?